### PR TITLE
feat: allow run on browser

### LINKED ISSUE #206: Headless bpmn executor
I'd like to use bpmn server in tiddlywiki, as discussed in https://github.com/bpmnServer/bpmn-server/issues/145#issuecomment-1852365807 some time ago, I hope this package can have a pure bpmn executor, that does not depend on mongodb and http server, so I can run the bpmn file purely in Typescript, and get result in Typescript, no HTTP request, also save the internal state to files or in-memory database, no mongodb requires.

Do you think this is possible?

### PR DESCRIPTION
**Pull request by linonetwo**
I inline the fs, path, process to the method that use them. Otherwise they will cause error even I already use a custom implementation.

'events' is nodejs only, so I install a tiny npm package.

I also fix all unused imports I meet, hope you don't mind.

After this, bpmn-server can run on browser, webworker, service worker, browser extensions. I think this make it possible to execute bpmn right inside bpmn-js based editor.

图片

basically will fixes #206 , tested locally with pnpm link. Can you release a new version after this?

### CODE REVIEW

**Comment by bpmn-ts** [comment]
Thanks, I will create a new branch with your changes soon.

Thanks

**Comment by linonetwo** [comment]
@ralphhanna Don't forget this! Without merge, other modification may cause git conflict.

**Comment by ralphhanna** [comment]
Please note I have already created a complete solution including:

1. New bpmn-server branch with minimum dependencies 
2. new bpmn-browser to work within the browser 

to run:
```sh
git clone -b RunInBrowser  https://github.com/bpmnserver/bpmn-server
git clone https://github.com/bpmnserver/bpmn-browser
cd bpmn-browser
npm i
```

**Comment by linonetwo** [comment]
I see, but I want to use it in Tiddlywiki, can you merge and release new bpmn-server version?

Also, should I close this PR?

**Comment by linonetwo** [comment]
The fs here may break the browser support?

https://github.com/bpmnServer/bpmn-browser/blob/573fe9f6160ce9979cd460e17703eb3a382d9915/src/WorkflowApp/appDelegate.ts#L7
